src/jit/hwintrinsicxarch.cpp

```diff
@@ -219,30 +219,20 @@ unsigned Compiler::simdSizeOfHWIntrinsic(NamedIntrinsic intrinsic, CORINFO_SIG_I
 }
 
 //------------------------------------------------------------------------
-// numArgsOfHWIntrinsic: get the number of arguments based on table and
-// if numArgs is -1 check number of arguments using GenTreeHWIntrinsic
-// node unless it is nullptr
+// numArgsOfHWIntrinsicSlow: gets the number of arguments for the hardware
+// intrinsic by "counting" the number of operands in 'node'. This function
+// is used as fallback in numArgsOfHWIntrinsic function if the table
+// entry is - 1
 //
 // Arguments:
-//    intrinsic -- id of the intrinsic function
 //    node      -- GenTreeHWIntrinsic* node with nullptr default value
 //
 // Return Value:
 //     number of arguments
 //
-int Compiler::numArgsOfHWIntrinsic(NamedIntrinsic intrinsic, GenTreeHWIntrinsic* node)
+int Compiler::numArgsOfHWIntrinsicSlow(GenTreeHWIntrinsic* node)
```

Comment:
I do not think we need two member functions of numArgsOfHWIntrinsic, please combine them to one.

Reply:
Yes, I thought there was a general consensus that we should have a single method.